### DISCUSSION TITLE
Nominal Types integration for Typebox

### DISCUSSION THREAD

**Discussion by castarco** [sinclairzx81/typebox#952] [Show and tell]
Hi! :) ,

these past 2 days I've been working on a small integration that lets us define schemas with brand/nominal types:

https://www.npmjs.com/package/@coderspirit/nominal-typebox

## Motivation

- I'm a geek 🤓 
- Use Cases:
  - Tracing values provenance (a "soft" form of compile-time values tainting)
  - Avoiding potential errors caused by parameters ordering, where the types alone are not enough to distinguish the parameters. This is not an hypothetical case, it happens in real life, and was in fact why I worked on this library.

## Context

Typebox is complex enough by itself, and adding this capability to Typebox would have been a daunting task.

I think that having this feature via accompanying libraries offers a good balance between flexibility and integration quality: Other people could implement similar integrations for their nominal/branding libraries of choice.

Long ago I tried to increase the popularity of my library `@coderspirit/nominal`, but I failed miserably.

I also tried to provide a package with common symbols (`@coderspirit/nominal-symbols`), so people would be able to create nominal/branding libraries that would be compatible with each other... and I also failed miserably.

Long story short: Given that we won't have a standard solution for nominal types for a long time, we have to craft specific accompanying libraries for each one of these solutions (in case we care about integrating them, of course, which I do for the case of my own library).

**Comment by sinclairzx81** [comment]
@castarco Hey! This is a cool project!

Would you like to send through a PR to include nominal in the TB ecosystem list? I think would be an cool addition, especially for users looking to work with various identifiers and string formats. Nominal looks really useful! 

Awesome work :)

**Comment by airtonix** [comment]
did this ever get introduced into typebox?
